torch/storage.py

```diff
@@ -156,10 +172,483 @@ def _new_shared(cls, size):
         else:
             return cls._new_using_fd(size)
 
+    def _untyped(self):
+        return self
+
+    @classproperty
+    def dtype(self):
+        return torch.uint8
 
 def _load_from_bytes(b):
     return torch.load(io.BytesIO(b))
 
 
 _StorageBase.type = _type  # type: ignore[assignment]
 _StorageBase.cuda = _cuda  # type: ignore[assignment]
+
+
+@lru_cache
+def _dtype_to_storage_type_map():
+    return {
+        torch.double: 'DoubleStorage',
+        torch.float: 'FloatStorage',
+        torch.half: 'HalfStorage',
+        torch.long: 'LongStorage',
+        torch.int: 'IntStorage',
+        torch.int16: 'ShortStorage',
+        torch.int8: 'CharStorage',
+        torch.uint8: 'ByteStorage',
+        torch.bool: 'BoolStorage',
+        torch.bfloat16: 'BFloat16Storage',
+        torch.cdouble: 'ComplexDoubleStorage',
+        torch.cfloat: 'ComplexFloatStorage',
+        torch.qint8: 'QInt8Storage',
+        torch.qint32: 'QInt32Storage',
+        torch.quint8: 'QUInt8Storage',
+        torch.quint4x2: 'QUInt4x2Storage',
+    }
+
+@lru_cache
+def _storage_type_to_dtype_map():
+    dtype_map = {
+        val: key for key, val in _dtype_to_storage_type_map().items()}
+    return dtype_map
+
+class TypedStorage:
+    is_sparse = False
```

Comment:
Huh, I didn't know this could be a thing

Reply:
Yeah, it's a bit weird. It's supposed to always be False though. I don't think (but not 100% sure) that it's actually being called anywhere for TypedStorage, but it's being called for ByteStorage within `torch._utils._type()`.

https://github.com/pytorch/pytorch/blob/e8762c2598e3f9666d7015f74b2b51a17fe6ef87/torch/_utils.py#L43

That function is kind of confusing to me.

I figured if ByteStorage has `is_sparse`, TypedStorage probably should as well?

Reply:
It would probably be profitable to do a pass over all of the Storage methods/properties and axe or deprecate the ones that aren't being used. Not here though.